modules/core/src/main/java/org/apache/ignite/internal/processors/cluster/GridClusterStateProcessor.java

```diff
@@ -662,6 +689,23 @@ private boolean isBaselineSatisfied(BaselineTopology blt, List<ClusterNode> srvN
             U.warn(log, "Received state finish message with unexpected ID: " + msg);
     }
 
+    /** */
+    public void onStateMessage() {
+        if (ctx.clientNode() && !ctx.isDaemon())
+            changeStateInProgress = new CountDownLatch(1);
+    }
+
+    /** */
+    public void onChangeStateMessageWasProcessed() {
```

Comment:
I suggest to rename this method to `onStateMessageProcessingFinished`.

Reply:
Done